docs/additional/internal_aspects_de.md

```diff
@@ -0,0 +1,313 @@
+
+##### Trivy
+
+Es ist wichtig die Sicherheitslücken in einem Docker-Image, bzw. dem Zielsystem zu identifizieren und zu beheben.
+Dafür bietet sich der Scanner Trivy an, welcher Sicherheitslücken im System identifiziert und priorisiert.
+Die `dogu-build-lib` bietet dafür eine [Komponente][trivy] an, die wie folgt verwendet werden kann, um das Image eines
+Dogus zu scannen.
+
+```groovy
+EcoSystem ecoSystem = new EcoSystem(this, "gcloud-credentials-id", "ssh-credentials-id")
+Trivy trivy = new Trivy(this, ecoSystem)
+
+stage('Trivy scan') {
+    trivy.scanDogu("/dogu", TrivyScanFormat.HTML, TrivyScanLevel.CRITICAL, TrivyScanStrategy.FAIL)
+}
+```
+
+[trivy]: https://github.com/cloudogu/dogu-build-lib/blob/develop/docs/development/trivy_de.md
+
+#### Manuelle Tests auf einer Test-Stage
+
+In der manuellen Testphase sollte das Dogu auf einer Test-Stage getestet werden.
+Dabei sollte die Test-Stage eine möglichst reale Umgebung abbilden.
+Das Dogu soll mit dem Befehl `cesapp install` installiert werden.
+Zusätzlich sollte es systematisch im Browser manuell getestet und als funktionstauglich eingestuft werden.
+Dabei sollte der Fokus auf den Features liegen, welche in Produktion die größte Relevanz haben.
+
+## Definition of Done
+
+Dieser Abschnitt beschreibt mögliche Voraussetzungen für einen finalen Zustand einer Entwicklung.
+Er schafft ein gemeinsames Verständnis für Entwickler, wann ein Feature fertig ist und kann verschiedenste Anforderungen
+beinhalten.
+
+Folgende Punkte beschreiben grundlegende Anforderungen, die als Basis einer Definition of Done dienen können.
+
+### Issue-Tracking
+
+- Für jede Änderung existiert ein Issue in einem Ticketsystem.
+- Es beschreibt das zu entwickelnde Produktinkrement.
+- Tickets aus öffentlichen Ticketsystem werden in internen Systemen verlinkt.
+
+### Dockerfile
+
+Anforderung für das Dockerfile sind [hier](interesting_aspects_de.md#container-images) zu finden.
+
+### Dogu.json
+
+Die Dogu.json muss folgende konfigurierbare Schlüssel enthalten:
+- `container_config/memory_limit`
+- `container_config/swap_limit`
+
+### Dogu-Scripting
+
+Anforderungen für Shell-Scripting sind [hier](interesting_aspects_de.md#bash-skripte) zu finden.
+
+### Dogu-Funktionalitäten
+
+- Backup & Restore-Fähigkeit betrachten
+- Upgrade-Fähigkeit betrachten
+    - Upgrade-Skripte müssen so gestaltet sein, dass aus allen alten Versionen des Dogus ein Upgrade möglich ist
+      (sofern das die im Dogu eingesetzte Software nicht verhindert)
+    - Ein Upgrade des Dogus von früherer Version ist zu testen
+- Admin-Gruppe
+    - Admin-Gruppe muss änderbar sein
+    - Admin-Gruppe aus dem `etcd` wird verwendet und bei einem Wechsel der Admin-Gruppe reagiert das Dogu entsprechend
+      darauf, dann wird der alten Admin-Gruppe das Admin-Recht entzogen. Siehe z.B. SonarQube und Nexus Repository
+      Manager.
+- Logging
+    - Log-Ausgabe des Dogus auf Fehler checken
+- Übernahme einer Änderung der CES-FQDN
```

Comment:
CES is an informal abbreviation and should be expanded. In this case I suggest the following because we already discussed this in https://github.com/cloudogu/dogu-development-docs/blob/main/docs/important/relevant_functionalities_de.md#%C3%A4nderbarkeit-der-fqdn

```suggestion
- Übernahme einer Änderung der FQDN
```